### PR TITLE
Part of #265 mypy - rework interval according to core

### PR DESCRIPTION
**Pull request by CoMPaTech**
Part of #265 
- updated `const` DEFAULT_SCAN_INTERVAL to match core
- `interval` was a `float` (while an `int`, while defined as `str`) :) (so mypy was actually on the ball)
- updated `# pw-beta` tags where due

### CODE REVIEW

File: custom_components/plugwise/config_flow.py [hunk 1]
```diff
@@ -334,27 +335,27 @@ async def async_step_init(
...
 
         coordinator = self.hass.data[DOMAIN][self.config_entry.entry_id][COORDINATOR]
-        interval = DEFAULT_SCAN_INTERVAL[coordinator.api.smile_type]
+        interval = DEFAULT_SCAN_INTERVAL[coordinator.api.smile_type]  # pw-beta
 
         data = {
             vol.Optional(
-                CONF_SCAN_INTERVAL,
+                timedelta(seconds=CONF_SCAN_INTERVAL),
                 default=self.config_entry.options.get(CONF_SCAN_INTERVAL, interval),
```
Comment (bouwew) [inline custom_components/plugwise/config_flow.py:343]: Shouldn't this be `default = timedelta(self.config_entry.options.get(CONF_SCAN_INTERVAL, interval)),`

Comment (CoMPaTech) [reply custom_components/plugwise/config_flow.py:343]: good catch, apparently not caught by flake/mypy

File: custom_components/plugwise/const.py [hunk 2]
```diff
@@ -40,13 +40,13 @@
...
-DEFAULT_MAX_TEMP: Final = 30
-DEFAULT_MIN_TEMP: Final = 4
-DEFAULT_PORT: Final = 80
-DEFAULT_SCAN_INTERVAL: Final[dict[str, float]] = {
-    "power": 10,
-    "stretch": 60,
-    "thermostat": 60,
+DEFAULT_MAX_TEMP = 30
+DEFAULT_MIN_TEMP = 4
+DEFAULT_PORT = 80
```
[the remark this answers is not in the document]
Comment (CoMPaTech) [reply custom_components/plugwise/const.py:45]: not intentional, merge error (apologies)

**Comment by bouwew** [comment]
DEFAULT_SCAN_INTERVAL is not used in any of the Plugwise Core files so it can be removed fully from Plugwise Core consts.py.
As a consequence, we can keep the code as it was, only change to:
```
DEFAULT_SCAN_INTERVAL: Final[dict[str, int]] = {
    "power": 10,
    "stretch": 60,
    "thermostat": 60,
}
```

**Comment by CoMPaTech** [comment]
> DEFAULT_SCAN_INTERVAL is not used in any of the Plugwise Core files so it can be removed fully from Plugwise Core consts.py. As a consequence, we can keep the code as it was, only change to:
> 
> ```
> DEFAULT_SCAN_INTERVAL: Final[dict[str, int]] = {
>     "power": 10,
>     "stretch": 60,
>     "thermostat": 60,
> }
> ```

https://github.com/home-assistant/core/blob/82af3146780b6f2054cd8e325acdff1623cefa1d/homeassistant/components/plugwise/coordinator.py#L31

File: custom_components/plugwise/config_flow.py (hunk 1 above)
Comment (CoMPaTech) [reply custom_components/plugwise/config_flow.py]: changed accordingly (and simplified)

**Comment by bouwew** [comment]
Yes, that's it. Now the question is, does that work when changing the interval via CONFIGURATION?

**Comment by CoMPaTech** [comment]
Intention is to try that tonight with this code :)

**Review by bouwew — APPROVED** [review]
Looking good!

**Pull request merged**